src/engine/N3Base/N3UIList.cpp

```diff
@@ -195,6 +195,17 @@ bool CN3UIList::SetString(int iIndex, const std::string & szString) {
     return false;
 }
 
+CN3UIString * CN3UIList::GetChildStrFromList(std::string str) {
+    for (std::list<CN3UIString *>::iterator it = m_ListString.begin(); it != m_ListString.end(); ++it) {
```

Comment:
This code is fine, and this is just a nitpick really, but this is a perfect use of `auto`, to not have to deal with the tedious templated type names, e.g.:
```cpp
    for (auto it = m_ListString.begin(); it != m_ListString.end(); ++it) {
```

I should also note that the preferable modern way to handle a loop like this is with their new for each loop syntax, e.g.:
```cpp
    for (CN3UIString* pString : m_ListString) {
```
Although in this particular case I don't know what this project is intending to support, since this is a newer language feature.

Reply:
Agree with @twostars here. The existing instances ideally we won't refactor in the same PR, just to make the PR with isolated changes, but when adding stuff, let's try and stick to modern C++.